### ISSUE TITLE
StatsdData not calling Flush on dispose

### ISSUE THREAD

**Issue by sandeep-eroad** [DataDog/dogstatsd-csharp-client#194]
Just wondering if StatsdData dispose should be calling Flush?

**Comment by rquackenbush** [comment]
I don't think it does. I am using this package in the context of a short lived application lifetime. I'm calling .Dispose(), but my telemetry isn't being transmitted reliably.

We're already wrapping the object, so I'm adding an explicit call to Flush for now as a workarond:

```c#
public void Dispose()
{
    if (!_disposed)
    {
        _dogStatsd.Flush();
        _dogStatsd.Dispose();
        _disposed = true;
    }
}
```
